### PR TITLE
python3Packages.molecule{,-docker}: init

### PR DESCRIPTION
**Pull request by ilpianista**
###### Motivation for this change

Package Molecule from Ansible.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by rnhmjoj** [comment]
@grahamcOfBorg build molecule molecule-docker

**Comment by rnhmjoj** [comment]
@GrahamcOfBorg build molecule molecule-docker

File: pkgs/development/python-modules/molecule-docker/default.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    docker
+    molecule
+    requests
+  ];
+
+  nativeBuildInputs = [
+    setuptools-scm
+  ];
+
+  checkPhase = "true";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/molecule-docker/default.nix:29]:
```suggestion
  dontCheck = false;
```

Why are those disabled? Please add an pythonImportsCheck instead.

File: pkgs/development/python-modules/molecule/default.nix [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    pyyaml
+    selinux-python
+    subprocess-tee
+  ];
+
+  nativeBuildInputs = [
+    setuptools-scm
+  ];
+
+  checkPhase = "true";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/molecule/default.nix:43]:
```suggestion
  dontCheck = false;
```

Why are those disabled? Please add an pythonImportsCheck instead.

File: pkgs/development/python-modules/molecule-docker/default.nix (hunk 1 above)
Comment (ilpianista) [reply pkgs/development/python-modules/molecule-docker/default.nix]: The tests needs `ansible-playbook` to run, but if I add it, then I get:

```
Found duplicated packages in closure for dependency 'ansible':
  ansible 2.9.26 (/nix/store/w2bi455gbz60j04k72qf3d0crjnd8am7-python3.8-ansible-2.9.26/lib/python3.8/site-packages)
  ansible 3.4.0 (/nix/store/izy6iilv0iv679slin9yzipfqrrkfdlr-python3.8-ansible-3.4.0/lib/python3.8/site-packages)

Package duplicates found in closure, see above. Usually this happens if two packages depend on different version of the same dependency.
```

File: pkgs/development/python-modules/molecule-docker/default.nix [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    docker
+    molecule
+    requests
+  ];
+
+  nativeBuildInputs = [
+    setuptools-scm
+  ];
+
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/molecule-docker/default.nix:29]: Why are tests disabled? Please add a comment.

Comment (ilpianista) [reply pkgs/development/python-modules/molecule-docker/default.nix:30]: Added.

**Comment by rnhmjoj** [comment]
With the latest `molecule` version the `molecule-docker` checkPhase is starting but the tests are still failing:
```
  ImportError: cannot import name 'lru_cache' from 'molecule.util' (/nix/store/w8wl95k9a0r3fspyg4pahn1yy8g1gpv3-python3.9-molecule-3.5.2/lib/python3.9/site-packages/molecule/util.py)
  
  
  ----------------------------------------------------------------------
  Ran 1 test in 0.007s
  
  FAILED (errors=1)
  Test failed: <unittest.runner.TextTestResult run=1 errors=1 failures=0>
  error: Test failed: <unittest.runner.TextTestResult run=1 errors=1 failures=0>
```

**Comment by rnhmjoj** [comment]
@GrahamcOfBorg build molecule molecule-docker

**Comment by rnhmjoj** [comment]
@ilpianista Thank you., it looks good now. Can you just squash the fixup commits?

**Comment by ilpianista** [comment]
@FRidh @jonringer friendly ping, thank you

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Please squash the commits into one per package and one for the maintainer addition.

**Comment by rnhmjoj** [comment]
ansible-compat is still failing to build and the packages are now out-of-date
I'm a bit tired of seeing this still not merged (no offence intended), so I've updated them, fixed the test and rebased unto master. Here's the [branch](https://github.com/rnhmjoj/nixpkgs/tree/init-molecule) with my changes.

**Comment by rnhmjoj** [comment]
@ilpianista Why close the pull request? You can pick up the [commits](https://github.com/rnhmjoj/nixpkgs/tree/init-molecule) and force push here. I could do that myself but I'd rather you verify the packages actually work first.

**Comment by ilpianista** [comment]
@rnhmjoj I'm sorry, but I thought that you were going to open a new PR from your branch.

**Comment by ilpianista** [comment]
@rnhmjoj it looks like I cannot re-open this PR. However, `molecule` does work fine! Thank you

**Comment by rnhmjoj** [comment]
Yes, it looks like github doesn't like if you delete the branch on your fork. Can you open a new PR?

**Pull request closed without merging**